parser.y

```diff
@@ -7374,6 +7467,26 @@ HashString:
 		$$ = $1
 	}
 
+RoleSpec:
+	Rolename
+	{
+		roleSpec := &ast.UserSpec{
```

Comment:
Should we add an `IsRole` field to the `UserSpec`?

Reply:
yes

Reply:
`auth.UserIdentity` is used for authorization, don't make it too heavy.

Role should not be used to login, so it's better to distinguish Role and User.

I think maybe you can define them like this:

```
// UserSpec is used for parsing create user statement.
type UserSpec struct {
	User    *auth.UserIdentity
	AuthOpt *AuthOption
    IsRole  bool
}
type RoleIdentity struct {
	Username     string
	Hostname     string
}
```

And you can still create the `UserSpec` here:

```
role := $1.(*auth.RoleIdentity)
&ast.UserSpec {
    User: &auth.UserIdentity{
	    Username: role.Username,
	    Hostname: role.Hostname,        
    },
    IsRole: true,
}
```

Reply:
got it